working/0723-static-extensions/feature-specification-variant1.md

```diff
@@ -0,0 +1,467 @@
+are always ignored by static member declarations, they are only relevant to
+constructors.*
+
+When a static extension declaration _D_ has no formal type parameters, and
+it has an on-type `C<S1 .. Sk>`, the instantiated constructor return type
+of _D_ is `C<S1 .. Sk>`. *In this case the on-type is a fixed type (also
+known as a ground type), e.g., `List<int>`. This implies that the
+constructor return type of D is the same for every call site.*
+
+Finally we have the general case: Consider a static extension declaration
+_D_ named `E` with formal type parameters `X1 extends B1 .. Xs extends Bs`
+and a constructor return type `C<S1 .. Sk>`. With actual type arguments
+`T1 .. Ts`, the instantiated constructor return type of _D_ with type
+arguments `T1 .. Ts` is `[T1/X1 .. Ts/Xs]C<S1 .. Sk>`.
+
+#### Explicit invocation of a constructor in a static extension
+
+Explicit constructor invocations are similar to static member invocations,
+but they need more detailed rules because they can use the formal type
+parameters declared by a static extension.
+
+An _explicitly resolved invocation_ of a constructor named `C.name` in a
+static extension declaration _D_ named `E` with `s` type parameters and
+on-class `C` can be expressed as `E<S1 .. Ss>.C.name(args)`,
+`E.C<U1 .. Uk>.name(args)`, or `E<S1 .. Ss>.C<U1 .. Uk>.name(args)`
+(and similarly for a constructor named `C` using `E<S1 .. Ss>.C(args)`,
+etc).
+
+A compile-time error occurs if the type arguments passed to `E` violate the
+declared bounds. A compile-time error occurs if no type arguments are
+passed to `E`, and type arguments `U1 .. Uk` are passed to `C`, but no list
+of actual type arguments for the type variables of `E` can be found such
+that the instantiated constructor return type of `E` with said type
+arguments is `C<U1 .. Uk>`.
+
+*Note that we must be able to choose the values of the type parameters
+`X1 .. Xs` such that the instantiated constructor return type is
+exactly `C<U1 .. Uk>`, it is not sufficient that it is a subtype thereof,
+or that it differs in any other way.*
+
+A compile-time error occurs if the invocation passes actual type arguments
+to both `E` and `C`, call them `S1 .. Ss` and `U1 .. Uk`, respectively,
+unless the instantiated constructor return type of _D_ with actual type
+arguments `S1 .. Ss` is `C<U1 .. Uk>`. In this type comparison, top types
+like `dynamic` and `Object?` are considered different, and no type
+normalization occurs. *In other words, the types must be equal, not
+just mutual subtypes.*
+
+*Note that explicitly resolved invocations of constructors declared in
+static extensions are an exception in real code, usable in the case where a
+name clash prevents an implicitly resolved invocation. However, implicitly
+resolved invocations are specified in the rest of this section by reducing
+them to explicitly resolved ones.*
+
+A constructor invocation of the form `C<T1 .. Tm>.name(args)` is partially
```

Comment:
Should this be a new section?  The section heading here is "Explicit invocation..." but this is now moving into implicit invocation.

Reply:
Right, I think it makes sense to just remove the word 'explicit' from the section title.